contracts/netmap/contract.go

```diff
@@ -416,11 +473,36 @@ func Netmap() []Node {
 // Current state of each node is represented in the State field. It MAY differ
 // with the state encoded into BLOB field, in this case binary encoded state
 // MUST NOT be processed.
+//
+// Deprecated: migrate to [ListCandidates].
 func NetmapCandidates() []Node {
 	ctx := storage.GetReadOnlyContext()
 	return getNetmapNodes(ctx)
 }
 
+// ListNodes provides an iterator to walk over current node set. It is similar
+// to [Netmap] method, iterator values are [Node2] structures.
+func ListNodes() iterator.Iterator {
+	return ListNodesEpoch(Epoch())
+}
+
+// ListNodesEpoch provides an iterator to walk over node set at the given epoch.
+// It's the same as [ListNodes] (and exposed as listNodes from the contract via
+// overload), but allows to query a particular epoch data if it's still stored.
```

Comment:
what result when data is not `still stored`? Such docs would help to distinguish empty netmaps from the no longer stored ones

Reply:
They're the same in the end, an iterator that returns nothing.

Reply:
then how user can determine whether netmap is empty or just unstored? Can we make a specific result for latter?

Reply:
result collision can easily raise misunderstaning. If i get empty netmap, i cannot determine is it really empty or not. Having split return, some code could merge both results into one if it doesnt care, and the other would remain as responsive as possible

Reply:
Contracts only return a single result. You can also rely on the result if your `epoch` is not too distant (`SnapshotCount`) from the current. You can also do real historic invocations if needed (requires archiving RPC node, but possible). And most of all, if data is gone then I don't see how practically it makes any difference from an empty map. When you're dealing with this most of the time you're trying to check if some node was a part of some container previously, if the map is empty for whatever reason the answer is "no" and that's it.

Reply:
all this does not change the fact, but as long as there are no problems it is acceptable